shell/browser/electron_browser_handler_impl.cc

```diff
@@ -0,0 +1,172 @@
+// Copyright (c) 2019 Slack Technologies, Inc.
+// Use of this source code is governed by the MIT license that can be
+// found in the LICENSE file.
+
+#include "shell/browser/electron_browser_handler_impl.h"
+
+#include <utility>
+
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/render_frame_host.h"
+#include "content/public/browser/render_process_host.h"
+#include "mojo/public/cpp/bindings/self_owned_receiver.h"
+
+namespace electron {
+ElectronBrowserHandlerImpl::ElectronBrowserHandlerImpl(
+    content::RenderFrameHost* frame_host)
+    : render_process_id_(frame_host->GetProcess()->GetID()),
+      render_frame_id_(frame_host->GetRoutingID()),
+      weak_factory_(this) {
+  content::WebContents* web_contents =
+      content::WebContents::FromRenderFrameHost(frame_host);
```

Comment:
Could this be `nullptr`?
1. if not, let's `DCHECK`
2. if so, we should either:
  - make sure we never call this with nullptr and `goto 1`, or
  - handle the case of `nullptr` gracefully.

I'm not sure this class is meaningful without a `WebContents` so we probably need to prevent `null` WebContents from reaching here.

Reply:
I don't think it would ever be null, so we should be able to DCHECK